sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -1452,7 +1460,18 @@ public boolean apply(@Nonnull String input) {
                   + " but received: " + returnType);
             }
           }
-          convertedOptions.put(entry.getKey(), MAPPER.convertValue(values, type));
+          JavaType[] params = type.findTypeParameters(returnType);
+          if (Collection.class.isAssignableFrom(returnType) && params.length == 1) {
```

Comment:
Now that you got the accurate typing information, I don't think you need this special case as Jackson will pass the parametric type to the deserializer and it should effectively do what your doing here.